### PR TITLE
made it so creating a schedule creates the subsequent sessions

### LINKED ISSUE #126: Sessions not being created when valid schedule is made
Sessions should be automatically made for the set day and time of a given schedule throughout the given period and currently that is not the case.

### PR DESCRIPTION
**Pull request by bam95899**
fixes #126

### CODE REVIEW

File: sciencelabs/schedule/__init__.py [hunk 1]
```diff
@@ -110,6 +111,17 @@ def create_schedule_submit(self):
             courses = form.getlist('courses')
             self.schedule.create_schedule(term, term_start_date, term_end_date, term_id, name, room,
                                                     start_time, end_time, day_of_week, leads, tutors, courses)
+
+            d1 = active_semester.startDate
+            d2 = active_semester.endDate
```
Comment (c-schwarze) [inline sciencelabs/schedule/__init__.py]: I don't love the d1 and d2 variable names.

File: sciencelabs/schedule/__init__.py [hunk 2]
```diff
@@ -110,6 +111,17 @@ def create_schedule_submit(self):
...
+            delta = d2 - d1
+            for i in range(delta.days + 1):
```
Comment (c-schwarze) [inline sciencelabs/schedule/__init__.py]: worth a comment for why `+ 1` is used.

File: sciencelabs/schedule/__init__.py [hunk 3]
```diff
@@ -110,6 +111,17 @@ def create_schedule_submit(self):
...
+                next_day_of_week = (d1 + timedelta(i)).weekday()
+                next_date = (d1 + timedelta(i))
+                if (next_day_of_week == day_of_week - 1 and next_day_of_week != 6) or next_day_of_week == 6 and day_of_week == 0:
```
Comment (c-schwarze) [inline sciencelabs/schedule/__init__.py]: Maybe you could use some modulus (%) here?

File: sciencelabs/schedule/__init__.py (hunk 1 above)
Comment (bam95899) [reply sciencelabs/schedule/__init__.py]: changed

**Comment by c-schwarze** [comment]
1 more thing. Would it make sense to just loop over the day of week, instead of everyday inbetween? The means only 1 in every 7 loops are actually used. Might be worth updating.

File: sciencelabs/schedule/__init__.py (hunk 2 above)
Comment (bam95899) [reply sciencelabs/schedule/__init__.py]: added comment

File: sciencelabs/schedule/__init__.py (hunk 3 above)
Comment (bam95899) [reply sciencelabs/schedule/__init__.py]: Im a little confused how modulus could be used here.

**Comment by bam95899** [comment]
I need to know what the exact date something is in order to create a session and am confused how I would do that when just looping over the day of the week.

File: sciencelabs/schedule/__init__.py (hunk 3 above)
Comment (c-schwarze) [reply sciencelabs/schedule/__init__.py]: I might be misunderstanding this logic. Lets talk this through next time you are in.

Comment (c-schwarze) [reply sciencelabs/schedule/__init__.py]: Also, if you are able to do the suggestion below, you probably won't need this if statement anymore (maybe).

**Comment by c-schwarze** [comment]
Well, I think you could replace `for i in range(delta.days + 1):` with something the loops over just the "sundays inbetween". Find the next sunday, use that as the starting for loop date. Then jump 1 week at a time until you are past the enddate.

Does that make sense?

**Comment by bam95899** [comment]
I have my initial loop ```            for i in range(7):
                next_day_of_week = (start + timedelta(i)).weekday()
                if (next_day_of_week == day_of_week - 1 and next_day_of_week != 6) or next_day_of_week == 6 and day_of_week == 0:
                    dow = i
                    break```

that sets the correct day of week that i will always occur on and then I pass it into another loop which only loops over every seventh day (every "Sunday" for example)
```            for i in range(dow, delta.days + 1, 7):
                next_date = (start + timedelta(i))
                self.session.create_session(active_semester.id, next_date, start_time, end_time, None, None,
                                            room, "", 0, name)```

File: sciencelabs/schedule/__init__.py (hunk 3 above)
Comment (bam95899) [reply sciencelabs/schedule/__init__.py]: Did the statement below

**Comment by bam95899** [comment]
It turns out that sessions weren't being made since they are only made when they are created for a date sooner than today and that is also why new schedules weren't showing up in the schedule section, since all the schedules were for dates that have already past.

**Pull request closed without merging**